File: eco-core/core-plugin/src/main/java/com/willfp/itemstats/stats/stats/StatItemDamage.java
package com.willfp.itemstats.stats.stats;

import com.willfp.itemstats.stats.Stat;
import com.willfp.itemstats.stats.util.StatChecks;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.player.PlayerItemDamageEvent;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public class StatItemDamage extends Stat {
    public StatItemDamage() {
        super("item_damage");
    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    public void statListener(@NotNull final PlayerItemDamageEvent event) {
        ItemStack itemStack = event.getItem();

        double value = StatChecks.getStatOnItem(itemStack, this);
        value += event.getDamage();
        StatChecks.setStatOnItem(itemStack, this, value);
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/stattrackers/stats/stats/StatItemDamage.kt
package com.willfp.stattrackers.stats.stats

import com.willfp.stattrackers.stats.Stat
import com.willfp.stattrackers.stats.incrementStatValue
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.player.PlayerItemDamageEvent

class StatItemDamage : Stat("item_damage") {
    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun statListener(event: PlayerItemDamageEvent) {
        val itemStack = event.item
        if (itemStack.type.maxStackSize > 1) {
            return
        }

        itemStack.incrementStatValue(this, event.damage.toDouble())
    }
}
